### DISCUSSION TITLE
Can we playback in a loop a section of a song?

### DISCUSSION THREAD

**Discussion by Djard** [hydrogen-music/hydrogen#1526] [Hydrogen on Windows]
I know I'm asking a lot questions, but I cannot find help in the manual.

1. Is there any way to remove an added Playback Track?

2. Any way to play back in a loop a section of a song, like selecting only four measures? I am able to marquee a section, but the marquee disappears.

**Comment by theGreatWhiteShark** [comment]
Hey @Djard ,

> I know I'm asking a lot questions, but I cannot find help in the manual.

Don't worry.

> Is there any way to remove an added Playback Track?

No. The UX of the playback track is still a little bit crude I have to admit. Currently you can either mute the playback track or load an empty sample/sound file.

> I am able to marquee a section, but the marquee disappears.

Do you mean the red rectangle appearing when right-clicking a section in the song ruler prior to the end of the song and moving the cursor to the right? This is a(n undocumented) stub introduced in the code at some point which just handles the visual side but does not change anything apart from that. I'll remove it in the upcoming release.

But looping itself is not supported in Hydrogen yet.

**Comment by Djard** [comment]
Thanks for the help.

I tried renaming the added audio file in its original location, hoping to detach it; but that caused Hydrogen to fail when attempting playback. A MS Visual C++ Runtime Library notice appears, stating "Assertion failed, C:\projects\hydrogen\src....Sampler.cpp Line:908. Expression: pSample." Undoing the rename resolved that issue.

The dashed line rectangle to which I refer is white, the one I use to copy and paste a group of patterns (see attached). It would be nice if the marqueed rectangle did not disappear until the copy & paste action is finished.

![Hydrogen Marquee](https://user-images.githubusercontent.com/99554465/158029369-7651d43f-dfef-4b2f-9800-204be29100bf.jpg)

BTW, Ctrl+Spacebar does not repeat playback from the point on the timeline at which the cursor was placed when playing a song. Any way to make the hotkey functional?

**Reply by theGreatWhiteShark** [reply]
> I tried renaming the added audio file in its original location, hoping to detach it; but that caused Hydrogen to fail when attempting playback. A MS Visual C++ Runtime Library notice appears, stating "Assertion failed, C:\projects\hydrogen\src....Sampler.cpp Line:908. Expression: pSample." Undoing the rename resolved that issue.

When renaming the file in a file browser, Hydrogen is not aware of the changes and still tries to read from the previous filename. Although, it would be nice if it just fails gracefully like muting the audio instead of an assertion fail. I'll check.

> The dashed line rectangle to which I refer is white, the one I use to copy and paste a group of patterns (see attached).

Yes. This selection is only used for copy&paste notes and patterns.

>  It would be nice if the marqueed rectangle did not disappear until the copy & paste action is finished.

Hmm. That might be a little bit misleading since top right corner of the pasted selection is not determined by the top-right corner of the selection triangle but the top and right-most pattern.

But I agree that the selection is a little hard to see depending on the colors you use in the song editor. In the upcoming version this will be replaced by a change in border color of the individual grid cells (black  = unselected, white = selected).

 
![selection_screenshot](https://user-images.githubusercontent.com/14164547/158051920-0b2f8ee1-9f0d-4165-aec5-08238ca757bd.png)

> BTW, Ctrl+Spacebar does not repeat playback from the point on the timeline at which the cursor was placed when playing a song. Any way to make the hotkey functional?

So, pressing <kbd>Ctrl</kbd> + <kbd>Space</kbd> does affect the playback of patterns and notes in Hydrogen but not the playback track? I can't reproduce this behavior for version 1.1.1

**Comment by Djard** [comment]
I closed Hydrogen and re-opened my project. Doing so enabled the hotkey: I can now place the cursor on the timeline then use Ctrl+Space to repeat playback from the specified point in the timeline. The hotkey lets me repeatedly play back any section of a song, like a loop.

Apparently letting the program sit idle for a long time causes the hotkey to fail (as well as loss of playback).

Thanks for lending a hand. You guys are great!

**Reply by theGreatWhiteShark** [reply]
> Apparently letting the program sit idle for a long time causes the hotkey to fail (as well as loss of playback).

hmm. This sounds like something is going wrong not within Hydrogen but on system side. Maybe even in the underlying library we are using to build the GUI. These things are very time consuming to debug/fix. Especially since it takes a large amount of time for the bug to show up and no one of us is a native Windows user. I'm afraid the chance for this being fixed is very slim.

> Thanks for lending a hand. You guys are great!

You are welcome
